crates/provider/src/fillers/nonce.rs

```diff
@@ -6,15 +6,89 @@ use crate::{
 use alloy_network::{Network, TransactionBuilder};
 use alloy_primitives::Address;
 use alloy_transport::{Transport, TransportResult};
+use async_trait::async_trait;
 use dashmap::DashMap;
 use futures::lock::Mutex;
 use std::sync::Arc;
 
-/// A [`TxFiller`] that fills nonces on transactions.
+/// A trait that determines the behavior of filling nonces.
+#[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
+#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
+pub trait NonceManager: Clone + Send + Sync + std::fmt::Debug + Default {
```

Comment:
hmm, do we need default here?

this could be a problem for nonce managers that require external context

Reply:
yes, it is really not used